app/src/main/kotlin/com/wire/android/ui/home/gallery/MediaGalleryScreen.kt

```diff
@@ -123,13 +125,17 @@ fun MediaGalleryScreen(mediaGalleryViewModel: MediaGalleryViewModel = hiltViewMo
 fun MediaGalleryContent(viewModel: MediaGalleryViewModel, mediaGalleryScreenState: MediaGalleryScreenState) {
     val context = LocalContext.current
     val uiState = viewModel.mediaGalleryViewState
-
+    val errorToastMessage = stringResource(R.string.label_no_application_found_open_downloads_folder)
     suspend fun showSnackbarMessage(message: String, actionLabel: String?, messageCode: MediaGallerySnackbarMessages) {
         val snackbarResult = mediaGalleryScreenState.snackbarHostState.showSnackbar(message = message, actionLabel = actionLabel)
         when {
             // Show downloads folder when clicking on Snackbar cta button
             messageCode is MediaGallerySnackbarMessages.OnImageDownloaded && snackbarResult == SnackbarResult.ActionPerformed -> {
-                context.startActivity(Intent(DownloadManager.ACTION_VIEW_DOWNLOADS))
+                try {
+                    context.startActivity(Intent(DownloadManager.ACTION_VIEW_DOWNLOADS))
+                } catch (e: ActivityNotFoundException) {
+                    Toast.makeText(context, errorToastMessage, Toast.LENGTH_SHORT).show()
```

Comment:
how about extracting the whole this peace of code into some `fun openDownloadFolder(context: Context)` in `util` package and use it everywhere we need?